src/TrueLayer/Serialization/OneOfJsonConverter.cs

```diff
@@ -27,6 +28,11 @@ public OneOfJsonConverter(OneOfTypeDescriptor descriptor, string discriminatorFi
 
             var doc = JsonDocument.ParseValue(ref reader);
 
+            if (doc.RootElement.ValueKind == JsonValueKind.Object && !doc.RootElement.EnumerateObject().Any())
+            {
+                return default;
+            }
```

Comment:
Sometimes in the `AutherizationFlow.Configuration` object from the `GetPaymentResponse`, we get empty objects for discriminated types like the `ProviderSelected` one:
```json
"authorization_flow": {
        "actions": {
            "next": {
                "type": "retry",
                "retry_options": [
                    "restart"
                ]
            }
        },
        "configuration": {
            "provider_selection": {},
            "redirect": {
                "return_uri": "http://localhost:3000/callback",
                "direct_return_uri": "http://localhost:3000/callback"
            },
            "retry": {}
        }
    }
```

This change forces the json discriminator converter to return the default value for that discriminated type.